src/BiDirectionalNodeImp.h

```diff
@@ -0,0 +1,51 @@
+//
+// Created by shvmo on 02/05/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_BIDIRECTIONALNODEIMP_H
+#define DATASTRUCTURESWET1_BIDIRECTIONALNODEIMP_H
+
+#include "BiDirectionalNode.h"
+
+template<class Type>
+BiDirectionalNode<Type>::BiDirectionalNode() = default;
+
+template<class Type>
+BiDirectionalNode<Type>::~BiDirectionalNode() = default;
```

Comment:
can be moved to header

Reply:
According to this: https://stackoverflow.com/questions/43733938/default-c-definitions-in-body-or-header-files
it's better to implement in the body.